pkg/image/builder/image_builder.go

```diff
@@ -80,30 +103,140 @@ func (b *ImageBuilder) Run(ctx context.Context, targetRef string, layoutPath lay
 	if err != nil {
 		return err
 	}
-	idxManifest, err := idx.IndexManifest()
+	// make a copy of the original manifest for later
+	originalIdxManifest, err := idx.IndexManifest()
+	if err != nil {
+		return err
+	}
+	originalIdxManifest = originalIdxManifest.DeepCopy()
+
+	// process the image index for updates to images discovered along the way
+	resultIdx, err := b.processImageIndex(ctx, idx, &v2format, update, targetRef, layers...)
```

Comment:
Nit: in the long run I'd suggest avoiding passing parameter which are modified by function, but rather explicitly return that value.